Code/GraphMol/FMCS/MaximumCommonSubgraph.cpp

```diff
@@ -446,13 +447,32 @@ void MaximumCommonSubgraph::makeInitialSeeds() {
   }
   size_t nq = QueryMolecule->getNumAtoms();
   for (size_t i = 0; i < nq; i++) {  // all query's atoms
+    const Atom *queryMolAtom = QueryMolecule->getAtomWithIdx(i);
+    bool isQueryMolAtomInRing = queryIsAtomInRing(queryMolAtom);
     unsigned matched = 0;
     for (std::vector<Target>::const_iterator tag = Targets.begin();
          tag != Targets.end(); tag++) {
       size_t nt = tag->Molecule->getNumAtoms();
       for (size_t aj = 0; aj < nt; aj++) {
         if (tag->AtomMatchTable.at(i, aj)) {
+          const Atom *targetMolAtom = tag->Molecule->getAtomWithIdx(aj);
+          bool isTargetMolAtomInRing = queryIsAtomInRing(targetMolAtom);
           ++matched;
+          if (!(Parameters.BondCompareParameters.CompleteRingsOnly &&
+            (isQueryMolAtomInRing || isTargetMolAtomInRing))) {
+            QueryMoleculeSingleMatchedAtom = (QueryMoleculeSingleMatchedAtom
```

Comment:
This construct is "a bit" too dense to be readable/maintainable.
Please use a normal if instead of the ternary operator and do something about the nested ternary operators inside the lambda so that it doesn't require quite so much mental effort to understand what it's doing.